### DISCUSSION TITLE
Table Copy from Oracle to SQL Server Generates Type Error

### DISCUSSION THREAD

**Discussion by jgriffin010** [sqlalchemy/sqlalchemy#6440] [Usage Questions] [answered]
I would like to copy a table structure in Oracle and create the same table in SQL Server.  I am using the following code to attempt this:

### **Code Fragment**
```
# Oracle (Seems to work fine)
engine1 = create_engine(
    cstr1, convert_unicode=False, pool_recycle=10, pool_size=50, echo=False, max_identifier_length=128)
workorder = Table('workorder', metadata1, autoload=True, autoload_with=engine1, schema='DEV')

# SQL Server Azure (Error Here on Table Create)
engine2 = engine = create_engine(
    "mssql+pymssql://username:password@db.database.windows.net:1111/DATA", echo=True)
table.metadata.create_all(engine2)
```

### **Error Stack Generated**
```
AttributeError: 'MSTypeCompiler' object has no attribute 'visit_NUMBER'
sqlalchemy.exc.UnsupportedCompilationError: Compiler <sqlalchemy.dialects.mssql.base.MSTypeCompiler object at 0x0000013CFC159F70> can't render element of type NUMBER 
sqlalchemy.exc.CompileError: (in table 'WORKORDER', column 'estmatcost'): Compiler <sqlalchemy.dialects.mssql.base.MSTypeCompiler object at 0x0000013CFC159F70> can't render element of type NUMBER
```

### **Question**
Should I be investigating the SQL Server driver or working through on the SQL Alchemy side?  Any suggestions on troubleshooting are much appreciated!

Thanks for having this support function available to the community!!

### **Version Information**
O/S: Windows 10 with Anaconda 2.0.1
Python: 3.8.8
sqlAlchemy: 1.4.7 (latest version in Anaconda)
cx_Oracle: 8.1.0

**Comment by zzzeek** [comment, marked as answer]
hi there -

look into https://docs.sqlalchemy.org/en/14/core/reflection.html#reflecting-with-database-agnostic-types for how to deal with this use case.

**Reply by jgriffin010** [reply]
This article did the trick.  With a modification of the Table schema property, the table create worked nicely.  Thanks for the guidance
